partiql-plan/build.gradle.kts

```diff
@@ -52,6 +52,7 @@ val generate = tasks.register<Exec>("generate") {
         "--poems", "builder",
         "--poems", "util",
         "--opt-in", "org.partiql.value.PartiQLValueExperimental",
+        "--restrict-interface-impl",
```

Comment:
Currently, restricting interface implementation using the annotation is only used for the `partiql-plan`